src/oogeso/core/devices/compressor.py

```diff
@@ -1,46 +1,50 @@
+from typing import Dict, List, Optional, Union
+
 import pyomo.environ as pyo
 
+from oogeso import dto
+from oogeso.core.devices.base import Device
 from oogeso.core.networks.network_node import NetworkNode
-from . import Device
-from oogeso.dto import (
-    CarrierData,
-    DeviceCompressor_elData,
-    DeviceCompressor_gasData,
-)
-from typing import Union
 
 
-class Compressor_el(Device):
-    "Electric compressor"
+class CompressorEl(Device):
+    """Electric compressor."""
+
     carrier_in = ["gas", "el"]
     carrier_out = ["gas"]
     serial = ["gas"]
 
-    def _rules(self, model, t, i):
+    def __init__(
+        self,
+        dev_data: Union[dto.DeviceCompressorGasData, dto.DeviceCompressorElData],  # Fixme: Correct?
```

Comment:
When it comes to redefining compute CO2 in each device class I would think it would be more robust since you have to define it for each device. But I kind of agree that it is a lot of duplication, so I will revert it for now.